src/pip/_internal/vcs/bazaar.py

```diff
@@ -41,23 +46,26 @@ def export(self, location, url):
 
         url, rev_options = self.get_url_rev_options(url)
         self.run_command(
-            ['export', location, url] + rev_options.to_args(),
+            ['export', location, hide_url(url)] + rev_options.to_args(),
             show_stdout=False,
         )
 
     def fetch_new(self, dest, url, rev_options):
```

Comment:
It looks like the PR would become simpler if you annotate some of the methods like these so that the `url` argument has to be `HiddenText`. (But not all. Methods like `get_url_rev_options()` and `resolve_revision()` you probably want to leave as accepting only `str`.) Then many of the function bodies won't need to change, and you can lean on the type-checker as opposed to tests for a lot of the correctness.